### PR TITLE
Sync with upstream

### PR DESCRIPTION
**Pull request by weseven**
Towards: https://github.com/giantswarm/giantswarm/issues/31529

Syncs with upstream chart v4.38.0 (node-exporter 1.8.2).

### CODE REVIEW

File: helm/node-exporter-app/templates/psp.yaml [hunk 1]
```diff
@@ -2,13 +2,14 @@
 apiVersion: policy/v1beta1
```
Comment (njuettner) [inline helm/node-exporter-app/templates/psp.yaml:2]: I'd add `{{- if and (le (int .Capabilities.KubeVersion.Minor) 24) (not .Values.global.podSecurityStandards.enforced) }}`

I don't think PSP is supported anymore after K8s 1.24

Comment (weseven) [reply helm/node-exporter-app/templates/psp.yaml:2]: Yep, makes a lot of sense!

File: helm/node-exporter-app/values.yaml [hunk 2]
```diff
@@ -316,12 +463,49 @@ readinessProbe:
...
+
+# Override version of app, required if image.tag is defined and does not follow semver
+version: ""
 
 disableConntrackCollector: false
 disableNvmeCollector: false
-
-global:
-  podSecurityStandards:
-    enforced: false
```
Comment (njuettner) [inline helm/node-exporter-app/values.yaml:327]: Do we wanna keep that? I saw we have it for mostly all of our apps, not sure about the impact if we remove it. WDYT?

File: helm/node-exporter-app/templates/psp.yaml (hunk 1 above)
Comment (nprokopic) [reply helm/node-exporter-app/templates/psp.yaml:2]: I would say "let's just remove the whole thing here", but considering that the whole process is manual and how we sync it from the upstream, every manual change would just make maintenance even harder.

So will leave that type of comments for some future node-exporter v2 or v3 plans 😄

File: helm/node-exporter-app/templates/verticalpodautoscaler.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{{- if and (.Capabilities.APIVersions.Has "autoscaling.k8s.io/v1") (.Values.verticalPodAutoscaler.enabled) }}
```
Comment (nprokopic) [inline helm/node-exporter-app/templates/verticalpodautoscaler.yaml:1]: I see this is coming from upstream, but just to comment - I sooo dislike these `.Capabilities.APIVersions.Has "<SOME API>"` checks in Helm charts.

It's a stateful logic in the Helm chart which can just bite us in the arse, as the chart can work differently on different occasions, depending on the state of the Kubernetes cluster in the moment of chart installation (and that state can change 5 seconds later if for some reason the CRD gets installed later).

But again, not something to change in this PR.

File: helm/node-exporter-app/values.yaml [hunk 4]
```diff
@@ -10,13 +14,103 @@ image:
...
+    enabled: false
+    ipFamilies: ["IPv6", "IPv4"]
+    ipFamilyPolicy: "PreferDualStack"
+  externalTrafficPolicy: ""
+
+# Set a NetworkPolicy with:
+# ingress only on service.port
+# no egress permitted
+networkPolicy:
+  enabled: false
```
Comment (nprokopic) [inline helm/node-exporter-app/values.yaml:113]: Should we enable this, or maybe add our own CiliumNetworkPolicy?

**Review by njuettner — APPROVED** [review]

**Pull request merged**